### PR TITLE
created form for verify customer

### PR DESCRIPTION
**Pull request by darmhoo**

### CODE REVIEW

File: src/components/modals/VerificationCheckList.vue [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+      vAgents: null,
+      verificationData: {},
+    }
+  },
+
+  methods: {
+    async fetchVerificationAgents() {
+      try {
+        this.$LIPS(true)
+        let res = await get('/api/get-users?role=21')
```
Comment (AltaraNg) [inline src/components/modals/VerificationCheckList.vue:299]: Do we still need this?

File: src/components/CustomerProfile.vue [hunk 2]
```diff
@@ -350,6 +365,27 @@ export default {
...
+          adaptive: true,
+          resizable: true,
+          draggable: true,
+          height: 'auto',
+          width: '50%',
+          clickToClose: true,
+        },
+        {
+          closed: event => {
+            console.log(event)
```
Comment (teebarg) [inline src/components/CustomerProfile.vue:384]: remove console

File: src/components/PaymentLog.vue [hunk 3]
```diff
@@ -889,49 +917,103 @@ export default {
...
+          });
+    },
+    watchSalesCategory() {
+      if (this.isAltaraPay) {
+        if (this.salesLogForm.sales_category_id == 9) {
+          this.salesLogForm.business_type_id = null
+          this.commitment.status = true
+          this.salesLogForm.repayment_duration_id = this.selectItem(this.repaymentDuration, "six_months")
+           this.salesLogForm.payment_type_id = this.selectItem(this.downPaymentRatesFiltered,  "twenty")
+           this.salesLogForm.payment_gateway_id = this.selectItem(this.paymentGateways, "paystack").id
```
Comment (teebarg) [inline src/components/PaymentLog.vue:938]: Add Optional chaining
`this.selectItem(this.paymentGateways, "paystack")?.id`

File: src/components/PaymentLog.vue [hunk 4]
```diff
@@ -889,49 +917,103 @@ export default {
...
+            //else return the rest
+            return !business_type.name.includes("No BS")
+          })
+        }
+      }
+    },
+    setSalesCategory() {
+      if (this.isAltaraCredit) {
+        this.salesCategories = this.salesCategories2.filter(obj => {
+          return obj.id !== 9
```
Comment (teebarg) [inline src/components/PaymentLog.vue:959]: Let's use `slug` here instead of `id`

File: src/components/PaymentLog.vue [hunk 5]
```diff
@@ -1142,10 +1231,41 @@ export default {
         this.rPayment = rePayment
         this.pPrice = total
         this.test1 = false
-        const months = this.rDuration / 30
 
+        //gray out button if not verified for NoBS
```
Comment (teebarg) [inline src/components/PaymentLog.vue:1235]: Will we still use this? if not delete

File: src/components/PaymentLog.vue [hunk 6]
```diff
@@ -1142,10 +1231,41 @@ export default {
...
+        // }
+
+        const months = this.rDuration / 30
         const cycle = Math.ceil(28 / this.repaymentCircle)
         const additionalRepayment = this.rPayment / (months * cycle)
+        if (this.isAltaraPay && this.salesLogForm.sales_category_id == 9) {
+          //check if on altara pay and New BS sales category
+
+          if (
+            this.salesLogForm.business_type_id.id == 16 ||
```
Comment (teebarg) [inline src/components/PaymentLog.vue:1252]: Let's use `slug` here instead of `id`

File: src/components/PaymentLog.vue [hunk 7]
```diff
@@ -1142,10 +1231,41 @@ export default {
...
+        //   ).name
+        //   if (salesCatName === "No BS" && !this.allowBSSale) {
+        //     throw new Error('Not Verified')
+        //   }
+        // }
+
+        const months = this.rDuration / 30
         const cycle = Math.ceil(28 / this.repaymentCircle)
         const additionalRepayment = this.rPayment / (months * cycle)
+        if (this.isAltaraPay && this.salesLogForm.sales_category_id == 9) {
```
Comment (teebarg) [inline src/components/PaymentLog.vue:1248]: Let's use `slug` here instead of `id`

File: src/components/PaymentLog.vue [hunk 8]
```diff
@@ -1142,10 +1231,41 @@ export default {
...
+            this.salesLogForm.business_type_id.id == 16 ||
+            this.salesLogForm.business_type_id.id == 15
+          ) {
+            //if biz-type is BS-new customer
+            this.commitment.percentage = 6
+            this.commitment.amount = this.selectedProduct.price * (6 / 100)
+
+            //add a 6% commision on the downpayment
+          } else if (
+            this.salesLogForm.business_type_id.id == 17 ||
```
Comment (teebarg) [inline src/components/PaymentLog.vue:1261]: Let's use `slug` here instead of `id`

File: src/components/PaymentLog.vue [hunk 9]
```diff
@@ -1171,12 +1291,13 @@ export default {
         //     icon: "error",
         //     title: "Plan is not available"
         // });
-        this.test1 = true
-        this.repaymentCircle = '0'
-        this.rDuration = '0'
-        this.fPayment = '0'
-        this.rPayment = '0'
-        this.pPrice = '0'
+       this.test1 = true
```
Comment (teebarg) [inline src/components/PaymentLog.vue:1294]: What is test1?

File: src/components/PaymentLog.vue [hunk 10]
```diff
@@ -1264,7 +1389,11 @@ export default {
     async getSalesCategory() {
       try {
         const fetchSalesCategory = await get(this.apiUrls.salesCategoryUrl)
+        this.salesCategories2 = fetchSalesCategory.data.data.data
```
Comment (teebarg) [inline src/components/PaymentLog.vue:1392]: use optional chaining

File: src/components/PaymentLog.vue [hunk 11]
```diff
@@ -1275,10 +1404,40 @@ export default {
       await get(`/api/sales-category/${salesCat}/roles`).then(res => {
         this.users = this.mergeArrays(res.data.data)
       })
+      let salesCatName = this.salesCategories.find(item => item.id === salesCat)
```
Comment (teebarg) [inline src/components/PaymentLog.vue:1407]: Add optional chaining incase of `null`

File: src/components/PaymentLog.vue [hunk 12]
```diff
@@ -1290,12 +1449,13 @@ export default {
       try {
         this.$LIPS(true)
         const fetchBusinessTypes = await get(this.apiUrls.businessTypes)
+        this.biz_type = fetchBusinessTypes.data.data.data
```
Comment (teebarg) [inline src/components/PaymentLog.vue:1452]: Add optional chaining

File: src/components/PaymentLog.vue (hunk 5 above)
Comment (darmhoo) [reply src/components/PaymentLog.vue:1235]: yes, in case we need to put the verification check again

File: src/components/PaymentLog.vue (hunk 9 above)
Comment (darmhoo) [reply src/components/PaymentLog.vue:1294]: that is for the checks before submit button is enabled

Comment (teebarg) [reply src/components/PaymentLog.vue:1298]: we should be using a better descriptive name

Comment (darmhoo) [reply src/components/PaymentLog.vue]: done

**Review by teebarg — APPROVED** [review]

**Pull request merged**